ast/shared/src/main/scala/jawn/ast/JValue.scala

```diff
@@ -254,13 +254,13 @@ case class DeferNum(s: String) extends JNum {
   lazy val n: Double = java.lang.Double.parseDouble(s)
 
   final override def getInt: Option[Int] = Some(n.toInt)
-  final override def getLong: Option[Long] = Some(util.parseLongUnsafe(s))
+  final override def getLong: Option[Long] = Some(n.toLong)
   final override def getDouble: Option[Double] = Some(n)
   final override def getBigInt: Option[BigInt] = Some(BigDecimal(s).toBigInt)
   final override def getBigDecimal: Option[BigDecimal] = Some(BigDecimal(s))
 
   final override def asInt: Int = n.toInt
-  final override def asLong: Long = util.parseLongUnsafe(s)
+  final override def asLong: Long = n.toLong
```

Comment:
Checking my understanding: right now it sometimes returns an incorrect value, and you want it to throw?  Changing behavior to start throwing scares me, particularly in a method not documented to throw.  But silently returning bad values isn't great, either.  And I guess it would align the behavior more closely with `.asInt`.

Reply:
There's a `util.parseLong(s)`, which isn't as fast as `util.parseLongUnsafe(s)`, but about 30% faster than `.toLong` and does throw on invalid input.  Maybe that would be a better implementation?